src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,22 @@
+package lotto.domain;
+
+import java.util.List;
+
+public class Lotto {
+    private final List<NumberGroup> numberGroups;
+
+    public Lotto(List<NumberGroup> numberGroups) {
+        this.numberGroups = numberGroups;
+    }
+
+    public void match(NumberGroup winningNumbers) {
+        for (NumberGroup numberGroup :
+                numberGroups) {
+            numberGroup.match(winningNumbers);
+        }
+    }
+
+    public List<NumberGroup> getNumberGroups() {
+        return numberGroups;
```

Comment:
컬렉션을 그대로 반환하면 변조를 할 수 있어 일급 컬렉션의 의미가 없어지는 것 같아요.
컬렉션을 수정하지 못하는 타입을 반환하면 어떨까요?
```suggestion
        return Collections.unmodifiableList(numberGroups);
```